jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenUtilTest.java

```diff
@@ -220,41 +222,31 @@ public String getBasedir() {
   }
 
   @Test
-  public void testCallMavenPluginWithGoal(@Mocked BuildPluginManager pluginManager)
-      throws PluginConfigurationException, MojoFailureException, MojoExecutionException, PluginManagerException {
-    // Given
-    MavenProject mavenProject = getMavenProject();
-    MavenSession mavenSession = getMavenSession();
-
-    // When
-    MavenUtil.callMavenPluginWithGoal(mavenProject, mavenSession, pluginManager,
-        "org.apache.maven.plugins:maven-help-plugin:help", log);
-
-    // Then
-    new Verifications() {
-      {
-        pluginManager.executeMojo(mavenSession, (MojoExecution) any);
-        times = 1;
-      }
-    };
+  public void testCallMavenPluginWithGoal() {
+    try (MockedConstruction<MojoExecutionService> mojoExecutionServiceMocked = Mockito.mockConstruction(MojoExecutionService.class)) {
```

Comment:
is it really necessary to have this try block ?

Reply:
@sunix Do you mean to completely remove the try block and write unit test case or take  mojoExecutionServiceMocked out from the try?
I have tried removing mojoExecutionServiceMocked out of try and placed it out, works fine too. Please clarify.Thanks!

Reply:
Mockito.mockConstruction **must** be used in a try-with-resources block or in a setup and teardown JUnit method to make sure that the static mocks are removed once the test completes.

Reply:
Yes, agreed. Gone through some resources and read about it. So I suppose that we need that try block, can I revert that back @manusa .

Reply:
yes